ci-scripts/arc/setup-runner-image.sh

```
@@ -0,0 +1,165 @@
+#
+# Binary URL resolution:
+#   When jq is available, this script queries ConsoleCLIDownload resources to find the
+#   exact binary download URLs for oc, kubectl, and virtctl that match the live cluster.
+#   These are passed to the Docker build as OC_URL, KUBECTL_URL, and VIRTCTL_URL build-args.
+#   If resolution fails (CRD not found, jq absent, etc.), the Dockerfile falls back to
+#   mirror.openshift.com / dl.k8s.io / GitHub releases using OC_VERSION / VIRTCTL_VERSION.
+
+set -euo pipefail
+ARC_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+CI_SCRIPTS_DIR="$(cd "${ARC_DIR}/.." && pwd)"
+source "${CI_SCRIPTS_DIR}/ci-tools.sh"
```

Shellcheck (0.11.0)

[info] Not following: ./ci-tools.sh was not specified as input (see shellcheck -x).

(SC1091)